test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Resources/DecimalFullMetadata.json

```diff
@@ -1,3 +1,3 @@
 ﻿{
-  "@odata.context":"http://localhost/$metadata#Edm.Decimal","@odata.type":"#Decimal","value":12345.99999
+  "@odata.context":"http://localhost/$metadata#Edm.Decimal","@odata.type":"#Decimal","value":12345
```

Comment:
Why was this change made?

Reply:
Guys, I think first we need to define expectation. Currently, we  have a Scale and Precision. So here is explanation at picture: 
![image](https://user-images.githubusercontent.com/50167116/154479210-8db97137-88e4-48b7-970e-5c2f995476dd.png)

In my opinion, first we need to validate: scale cannot be higher then precision, because it does not make sense 
I created this unit test for some use cases:

```
        [TestCase(3, 0, 123, "123")]
        [TestCase(7, 5, 123, "12.30000")]
        [TestCase(3, 2, 123, "1.23")]
        [TestCase(6, 3, 123, "123.000")]
        [TestCase(6, 3, 123.123, "123.123")]
        [TestCase(6, 4, 123.123, "12.3123")]
        public void DecimalConverterTest(int precision, int scale, decimal decimalValue, string expectedDecimalString)
        {
            var converter = new MyDecimalConverter(scale, precision);
            var actualDecimal = converter.ConvertDecimal(decimalValue).ToString();
            Assert.That(actualDecimal, Is.EqualTo(expectedDecimalString));
        }
```
Since it is not working currently, this change make surprise some people after updating package. I would be careful on making this change because it will affect highly in some use cases. For example, from our use case, we have Amount in transaction which we use Scale and Precision. So, our client may see different amount of transaction. 
So, it should be included in documentation and in release notes also.

Reply:
@komdil, in the test cases you've shared above, I have some confusions: Is the converter converting 123 to 1.23 when precision is 3 and scale is 2? Doesn't that actually change the value of the decimal? Same concern for 123.123 -> 12.3123.

Reply:
@habbes Actually, I don't know what is expected here. Look at the picture above, when precision is 3 and scale is 2 and decimal value is 1 it will be 1.00 and it looks OK. So, in the case when the decimal value is 123 I don't know what would be expected. Do you have any idea? What did you decide in your stand-up meeting?

Reply:
In my opinion, in OData having only scale without precision is make sense. Scale and precision are mostly will be used in database

Reply:
@komdil my view is that OData can either:
-  correct data to ensure it matches the specification of the Precision and Scale values
- or it can validate values check whether they match the specification and throw an exception if they don't
- or both

I think the second case is easier to deal with. We look at the input value and check whether it matches the expected precision and scale without trying to do any rounding or modification. In this scenario, it is the responsibility of the service owner or client to ensure the values are valid before serialization. Would this be feasible for you?

But if we want to support the request scenario were OData actual "coerces" the values to match the specified Precision and Scale, we should be aware that the input values might be impossible to represent with the required precision+scale combination. How do we handle such cases? I don't think we should try to force the values to fit the precision when that would effectively change the value of the input (e.g. transforming 123.123 to 12.123). In this case either OData should throw an exception, or it should do a best-effort and let the values through even if they exceed the expected precision.

Because of these considerations, I agree that it's better to just support Scale for now where we don't worry about precision. That way we only focus on rounding. We can deal with Precision when support for it is required.